### PR TITLE
[iOS] Add basic DisplayP3 color support

### PR DESCRIPTION
**Pull request by ryanlntn**
## Summary:

This adds initial support for wide gamut (DisplayP3) colors to React Native iOS per the [RFC](https://github.com/react-native-community/discussions-and-proposals/pull/738). It provides the ability to set the default color space to sRGB or DisplayP3 and provides the native code necessary to support `color()` function syntax per the [W3C CSS Color Module Level 4](https://www.w3.org/TR/css-color-4/#color-function) spec. It does _not_ yet support animations and requires additional JS code before fully supporting the `color()` function syntax. 

## Changelog:

[IOS] [ADDED] - Add basic DisplayP3 color support

## Test Plan:

Screenshot_20240131-100112

Follow test steps from #42831 to test support for `color()` function syntax.

To globally change the default color space to DisplayP3 make the following changes to RNTester AppDelegate.mm:
```diff
+ #import <React/RCTConvert.h>

   - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
   {
     // ...
+    RCTSetDefaultColorSpace(RCTColorSpaceDisplayP3);
     return [super application:application didFinishLaunchingWithOptions:launchOptions];
   }
```

### CODE REVIEW

File: packages/react-native/React/Base/RCTConvert.mm [hunk 1]
```diff
@@ -878,6 +880,38 @@ + (UIEdgeInsets)UIEdgeInsets:(id)json
   return names;
 }
 
+static RCTColorSpace defaultColorSpace = (RCTColorSpace)facebook::react::getDefaultColorSpace();
```
Comment (cipolleschi) [inline packages/react-native/React/Base/RCTConvert.mm:883]: this static property seems to be unused now that we fetch the value from the C++ layer.

File: packages/react-native/React/Base/RCTConvert.mm [hunk 2]
```diff
@@ -878,6 +880,38 @@ + (UIEdgeInsets)UIEdgeInsets:(id)json
...
+RCTColorSpace RCTGetDefaultColorSpace(void)
+{
+  return (RCTColorSpace)facebook::react::getDefaultColorSpace();
+}
+void RCTSetDefaultColorSpace(RCTColorSpace colorSpace)
+{
+  facebook::react::setDefaultColorSpace((facebook::react::ColorSpace)colorSpace);
+}
+
++ (UIColor *)createColorFrom:(CGFloat)r green:(CGFloat)g blue:(CGFloat)b alpha:(CGFloat)a
```
Comment (cipolleschi) [inline packages/react-native/React/Base/RCTConvert.mm:893]:
```suggestion
+ (UIColor *)RCTCreateColorWith:(CGFloat)red green:(CGFloat)green blue:(CGFloat)blue alpha:(CGFloat)alpha
```
in this way we keep the same syntax used by Apple, to be less confusing, but with the RCTPrefix we make sure to convey that it is a React Native method.

**Review by cipolleschi — CHANGES_REQUESTED** [review]
There are a couple of nits, plus CI is not very happy. Seem like that dynamic frameworks don't have access to the symbols...
You can debug it by running RNTester installing the pods with:
`USE_FRAMEWORKS=dynamic RCT_NEW_ARCH_ENABLED=1 bundle exec pod install`

The failure is likely due to some missing search paths in the React-Core.podspec.
We might need to add this line:
```
add_dependency(s, "React-graphics", :additional_framework_paths => ["react/renderer/graphics/platform/ios"])
```

[here](https://github.com/facebook/react-native/blob/main/packages/react-native/React-Core.podspec#L137)

**Comment by ryanlntn** [comment]
@cipolleschi Thanks that worked after I also updated React-graphics to only pull in React-jsi instead of all of React-Core.

**Review by cipolleschi — APPROVED** [review]
Amazing, thank you so much for this contribution @ryanlntn

**Pull request closed without merging**